site/website/blog/2020-09-18-cdc-meets-stream-processing.md

```diff
@@ -0,0 +1,178 @@
+effectively requires re-architecting existing systems to event-driven
+architectures and introducing several new components. This process is
+not always straightforward and also requires a shift in mindset.
+
+In this context, the functionality provided by change data capture
+technologies, for which Debezium is one of the, if not THE best
+open-source alternative, is a godsend. To be able to ingest data from
+relational databases, without affecting the applications that use them,
+changes the game for streaming systems. It becomes possible to safely
+extend old systems with all kinds of new functionality: real-time
+analytics, complex event processing, anomaly & fraud detection and so
+on.
+
+## Integration
+
+When we first considered integrating Debezium into Jet, the most
+important decisions were centered around the fact that Debezium is
+designed to be deployed via Apache [Kafka
+Connect](https://kafka.apache.org/documentation/#connect), which then
+takes care of *fault tolerance* and *scalability*. Fortunately, Jet is
+fully capable of providing these crucial services. Also, Kafka Connect
+is a good enough abstraction that we were able to mimic it for Debezium.
+
+We are aware that Debezium also offers an *embedded mode* for
+applications not interested in fault-tolerance guarantees such as
+exactly-once processing and resilience, but since Jet does not have a
+“dumbed down”version (even as full-blown is light enough to be
+embedded), we quickly discarded this approach.
+
+So, first, we added generic support for Kafka Connect sources to Jet,
+which should be a valuable feature even outside the scope of CDC. Then
+we’ve used Debezium to build a Kafka Connect source for Jet. Well…
+“build” might be overstating it. Debezium already is a Kafka Connect
+source. We just had to make sure that Jet’s specific fault-tolerance
+mechanisms will interact with it properly, through the Kafka Connect
+API.
+
+## Synergy
+
+One immediate benefit that Jet offers to Debezium users is eliminating
+the need for *external services*. No Zookeeper, no Kafka needed. When
+using Debezium through Jet, the latter takes care of the whole lifecycle
+and fault tolerance of all the components involved. The setup is greatly
+simplified.
+
+Then, obviously, there is the *stream processing capability*, because
+that’s what Jet does. Not only do you get access to the data, but you
+also have the toolbox to process it, extract whatever insights you need
+from it.
+
+In addition, Jet also aims to offer *further convenience* wrappers when
+the Debezium source is being used. For example:
+
+* builders for the most common configuration properties to make setting
+  up Debezium for some specific DB as simple as possible
+* standard Java interfaces to give structure to the complex Debezium
+  events
+* JSON parsing, including mapping to Objects, based on [Jackson
+  jr](https://github.com/FasterXML/jackson-jr), to simplify how parts
```

Comment:
```suggestion
  Jr](https://github.com/FasterXML/jackson-jr), to simplify how parts
```

Reply:
No, it's a small "j": https://github.com/FasterXML/jackson-jr
I don't like it either, but it's how it is.